apps/api/src/domain/services/query-services/base-query.service.ts

```diff
@@ -170,6 +161,6 @@ export abstract class BaseQueryService<
         return {
             data,
             actions: this.getInfoForIndexScopedCommands(),
-        } as unknown as AggregateIndexQueryResult<ViewModelWithTags<UViewModel>>;
+        } as unknown as IIndexQueryResult<ViewModelWithTags<UViewModel>>;
```

Comment:
All casting is permanent upstream, which is why we try to avoid it if possible. Sometimes, we just simply know more than the compiler. In the case here, I forgot to use the generic on `mixTagsIntoViewModel`, which is now called as
```ts
  data: mixTagsIntoViewModel<UViewModel>(...)
```
This allowed me to remove the cast and simply flow through the top-level generic `UViewModel` type.